module/VuFind/tests/unit-tests/src/VuFindTest/ILS/ConnectionTest.php

```diff
@@ -0,0 +1,207 @@
+                        'endDate' => date('Y-m-d', strtotime('now + 1 days')),
+                    ],
+                ],
+                true,
+            ],
+            'inside recurring limits' => [
+                [
+                    'Renewals' => [
+                        'recurringStart' => date('H:i', strtotime('now - 1 hours')),
+                        'recurringEnd' => date('H:i', strtotime('now + 1 hours')),
+                    ],
+                ],
+                true,
+            ],
+            'outside recurring limits' => [
+                [
+                    'Renewals' => [
+                        'recurringStart' => date('H:i', strtotime('now + 1 hours')),
+                        'recurringEnd' => date('H:i', strtotime('now - 1 hours')),
+                    ],
+                ],
+                false,
+            ],
+            'recurring crossing midnight' => [
+                [
+                    'Renewals' => [
+                        'recurringStart' => date('H:i', strtotime('now + 2 hours')),
+                        'recurringEnd' => date('H:i', strtotime('now + 1 hours')),
+                    ],
+                ],
+                true,
+            ],
+            'empty configuration' => [
+                [],
+                false,
+            ],
+            'startDate and endDate in the past' => [
+                [
+                    'Renewals' => [
+                        'startDate' => date('Y-m-d', strtotime('now - 2 days')),
+                        'endDate' => date('Y-m-d', strtotime('now - 1 days')),
+                    ],
+                ],
+                false,
+            ],
+            'startDate after endDate' => [
+                [
+                    'Renewals' => [
+                        'startDate' => date('Y-m-d', strtotime('now - 1 days')),
+                        'endDate' => date('Y-m-d', strtotime('now - 2 days')),
+                    ],
+                ],
+                false,
+            ],
+            'invalid values' => [
+                [
+                    'Renewals' => [
+                        'startDate' => 'testing string',
+                        'endDate' => 'true',
+                        'recurringStart' => 'starting',
+                        'recurringEnd' => 'ending',
+                    ],
+                ],
+                false,
+            ],
+        ];
+    }
+
+    /**
+     * Test that methods are blocked correctly according to configuration
+     *
+     * @param array $timedBlocks    timedBlocks as defined in Demo.ini
+     * @param bool  $expectedResult The expected result
+     *
+     * @dataProvider isMethodBlockedProvider
+     *
+     * @return void
+     */
+    public function testIsMethodBlocked(array $timedBlocks, bool $expectedResult): void
+    {
+        $this->setTimedBlocks($timedBlocks);
+        $this->assertEquals($this->connection->isMethodBlocked('Renewals'), $expectedResult);
```

Comment:
The expected value should come first to ensure appropriate messages in case of failure:

```suggestion
        $this->assertEquals($expectedResult, $this->connection->isMethodBlocked('Renewals'));
```